### PR TITLE
Update wicg-file-system-access for TS 4.6

### PR DESCRIPTION
**Pull request by sandersn**
Typescript 4.6 beta adds support for FileSystemHandle. This PR makes the existing Definitely Typed types compatible with it.

The DOM version is missing some features.

1. `FileSystemDirectoryHandle.getSystemDirectory` is removed. It was previously deprecated, so I don't think this is a problem.
2. FileSystemHandle is now a base class, not a union. If you want to narrow, you'll need to declare variables with type `FileSystemFileHandle | FileSystemDirectoryHandle`. I added a type alias for the union called `FileSystemHandleUnion`, which is a so-so name at best.

I tested this change on VS Code's source, which required a few places to changes to type annotations.

### CODE REVIEW

**Comment by RReverser** [comment]
You might want to coordinate with #58389 which was created a little earlier.

**Comment by sandersn** [comment]
Good idea. I actually think I'm going to go back to the DOM and see if I can add the correct `kind` subtypes to FileSystem(File|Directory)Handle before I finish this PR. That way narrowing will work even without this package.

**Comment by RReverser** [comment]
Yeah that's one thing I suggested in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58389#issuecomment-1020411470 too.

**Comment by sandersn** [comment]
https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1251 -- if @saschanaz likes it then I'll pull it over into the Typescript repo in the next day or two.

**Comment by sandersn** [comment]
The DOM update will be in the typescript nightly build. I'll update this PR tomorrow.

**Comment by sandersn** [comment]
I'm going to merge this since it now matches the DOM. We can update it further if we want to add features.

File: types/wicg-file-system-access/index.d.ts [hunk 1]
```diff
@@ -97,35 +98,50 @@ class FileSystemWritableFileStream extends WritableStream {
...
+    readonly kind: 'directory';
     getDirectoryHandle(name: string, options?: FileSystemGetDirectoryOptions): Promise<FileSystemDirectoryHandle>;
+    getFileHandle(name: string, options?: FileSystemGetFileOptions): Promise<FileSystemFileHandle>;
     removeEntry(name: string, options?: FileSystemRemoveOptions): Promise<void>;
     resolve(possibleDescendant: FileSystemHandle): Promise<string[] | null>;
-
     keys(): AsyncIterableIterator<string>;
-    values(): AsyncIterableIterator<FileSystemHandle>;
-    entries(): AsyncIterableIterator<[string, FileSystemHandle]>;
+    values(): AsyncIterableIterator<FileSystemDirectoryHandle | FileSystemFileHandle>;
```
Comment (RReverser) [inline types/wicg-file-system-access/index.d.ts:127]: You defined a (renamed) alias above, why not use it here?

**Comment by RReverser** [comment]
Looks like this was a breaking change for Chrome <=85 because it removes `getSystemDirectory`, which was covered by the test also removed in this PR :) https://github.com/GoogleChromeLabs/wasi-fs-access/issues/7

I appreciate it's a deprecated method, but I think it should've been kept with `@deprecated` tag like other methods in those typings.

@sandersn could you please add it back?

**Pull request merged**